akka-stream/src/main/scala/akka/stream/impl/fusing/Interpreter.scala

```diff
@@ -630,16 +639,27 @@ private[akka] class OneBoundedInterpreter(ops: Seq[Stage[_, _]],
       (pipeline(op): Any) match {
         case b: BoundaryStage ⇒
           b.context = new EntryState("boundary", op)
+          b.preStart(LifecycleContext)
```

Comment:
agree completely, sadly I wasn't able to get it working yet. Would be great to get some help from you when you're back (monday?).

My modifications were mostly focused on setting the context to Failed. But then I wasn't sure where to make it run to propagate the error. I guess I want to do this after all stages have been initialised, and then propagate the onError from my failed stage right away. I wasn't able to grasp how to express this in the interpreter sadly - so your insights would be very helpful here :-)